HumanControls/JoystickController.h

```diff
@@ -0,0 +1,31 @@
+#ifndef JOYSTICKCONTROLLER_H
+#define JOYSTICKCONTROLLER_H
+
+#include "Arduino.h"
+
+class JoystickController
+{
+public:
+    //deadZoneSize is the total proportion of the joystick's range of values to omit
+    JoystickController(unsigned int xPin,
+                       unsigned int yPin,
+                       unsigned int xDeadZoneSize,
+                       unsigned int yDeadZoneSize,
+                       unsigned int xMin,
+                       unsigned int xMax,
+                       unsigned int yMin,
+                       unsigned int yMax);
+    void update();
+    double getX();
+    double getY();
+
+private:
+    double calcX(int rawValue);
+    double calcY(int rawValue);
+
+    int xPin, yPin, xMin, xMax, yMin, yMax;
+    double x, y, xDeadMin, xDeadMax, yDeadMin, yDeadMax, xCenter, yCenter,
+        xHalfTotal, xOffset, yHalfTotal, yOffset;
```

Comment:
There are a lot of variables here that will take up a lot of memory when they could be calculated each time fairly easily. I think maybe we should just calculate some of this when we need it.

Reply:
Done